__tests__/components/ui/button.test.tsx

```
@@ -0,0 +1,90 @@
+
+  it('should merge custom className with default classes', () => {
+    render(<Button className="custom-class">Custom</Button>);
+    const button = screen.getByRole('button', { name: 'Custom' });
+    expect(button).toHaveClass('custom-class');
+    expect(button).toHaveClass('bg-primary');
+  });
+
+  it('should render as child component when asChild is true', () => {
+    render(
+      <Button asChild>
+        <a href="#">Link Button</a>
```

Biome (1.9.4)

[error] Provide a valid value for the attribute href.

The href attribute should be a valid a URL
Check this thorough explanation to better understand the context.

(lint/a11y/useValidAnchor)